classes/queued_table.php

```diff
@@ -0,0 +1,129 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace local_clonecategory;
+
+use table_sql;
+
+defined('MOODLE_INTERNAL') || die;
+require_once($CFG->libdir . '/tablelib.php');
+
+/**
+ * Clone category queued clones table.
+ *
+ * @package    local_clonecategory
+ * @copyright  2023 Matthew Hilton <matthewhilton@catalyst-au.net>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class queued_table extends table_sql {
+    /**
+     * Create table
+     * @param string $uniqueid
+     */
+    public function __construct(string $uniqueid) {
+        global $PAGE;
+
+        parent::__construct($uniqueid);
+
+        $columns = [
+            'id',
+            'timecreated',
+            'course',
+            'categoryfrom',
+            'categoryto',
+            'timestarted'
+        ];
+
+        $headers = [
+            get_string('queued:id', 'local_clonecategory'),
+            get_string('queued:timecreated', 'local_clonecategory'),
+            get_string('queued:course', 'local_clonecategory'),
+            get_string('queued:categoryfrom', 'local_clonecategory'),
+            get_string('queued:categoryto', 'local_clonecategory'),
+            get_string('queued:timestarted', 'local_clonecategory'),
+        ];
+
+        $this->define_columns($columns);
+        $this->define_headers($headers);
+        $this->baseurl = $PAGE->url;
+        $this->set_sql('id,customdata,timecreated,timestarted', '{task_adhoc}', "classname = :classname",
+            ['classname' => '\local_clonecategory\task\clone_course_task']);
+        $this->sortable(true, 'timecreated', SORT_DESC);
```

Comment:
the time created is gonna be random, once we join to an existing list of source courses I think we should sort by something unchanging, either the course name or its course idnumber. Leaning towards because that is what everything else is leveraging

Reply:
Can't really sort by course or category directly here since its contained inside the customdata col.

Instead I just made it sort by the customdata, which is indirectly sorting by the two and won't change.